classes/hypeJunction/Scraper/ScraperService.php

```diff
@@ -311,21 +318,18 @@ public function parseThumbs(array $data = []) {
 		$icons = (array) elgg_extract('icons', $data, []);
 
 		// Try 3 images and choose the one with highest dimensions
-		$thumbnails = $thumbnails + $icons;
-		$thumbs_parse = 0;
+		$thumbnails = array_filter(array_unique($thumbnails + $icons));
+		$thumbs_parsed = 0;
 		foreach ($thumbnails as $thumbnail) {
-			if ($thumbnail == $url) {
-				continue;
-			}
 			$thumbnail = elgg_normalize_url($thumbnail);
-			if (filter_var($thumbnail, FILTER_VALIDATE_URL)) {
-				$asset = $this->parse($thumbnail, $flush, false);
+			if (filter_var($thumbnail, FILTER_VALIDATE_URL) && $this->isImage($thumbnail)) {
```

Comment:
You can remove both checks now. I added image validation to https://github.com/hypeJunction/http-parser/commit/84f613645efcf67616248c23456140f19e643ad8. If URL is not valid, it will fail to parse, so no need to check either.